lib/r10k/cli/deploy.rb

```diff
@@ -23,6 +23,8 @@ def self.command
 
         required :c, :config, 'Specify a configuration file'
 
+        required nil, :cachedir, 'Specify a cachedir, overriding the value in config'
```

Comment:
Adding the `cachedir` command line flag here means that this will be a one off for the `r10k deploy environment` command, meaning that it won't work with `r10k deploy module` or `r10k puppetfile`. I know that this feature is being implemented for a specific use case but I dislike having one offs or inconsistencies unless there's a good reason for them. Does it make sense to keep this option limited to the single command?